apps/color-picker/src/locations/Field.tsx

```diff
@@ -0,0 +1,136 @@
+/** @jsxImportSource @emotion/react */
+
+import { FieldExtensionSDK } from '@contentful/app-sdk';
+import { Button, Flex, Form, Menu } from '@contentful/f36-components';
+import { ChevronDownIcon } from '@contentful/f36-icons';
+import tokens from '@contentful/f36-tokens';
+import { useFieldValue, useSDK } from '@contentful/react-apps-toolkit';
+import { css } from '@emotion/react';
+import { useEffect, useRef, useState } from 'react';
+import { v4 as uuidv4 } from 'uuid';
+
+// Dropdown + margin top + box shadow
+const HEIGHT_DEFAULT = 41;
+const HEIGHT_BASE = 52 + 4 + 14;
+const HEIGHT_ITEM = 36;
+
+const Field = () => {
+  const sdk = useSDK<FieldExtensionSDK>();
+  const [isOpen, setIsOpen] = useState(false);
+  const [value, setValue] = useFieldValue<Color>(
+    sdk.field.id,
+    sdk.field.locale
+  );
+  const customColorPicker = useRef<HTMLInputElement>(null);
+
+  useEffect(() => {
+    if (!isOpen) {
+      sdk.window.updateHeight(HEIGHT_DEFAULT);
+      return;
+    }
+
+    // @ts-ignore
+    const customItemHeight = sdk.parameters.instance.withCustomValue
+      ? HEIGHT_ITEM
+      : 0;
+    const calculatedHeight =
+      HEIGHT_BASE + customItemHeight + theme.colors.length * HEIGHT_ITEM;
+
+    sdk.window.updateHeight(calculatedHeight <= 400 ? calculatedHeight : 400);
+  }, [isOpen]);
+
+  // @ts-ignore
+  const theme = sdk.parameters.installation.themes[0];
+
+  return (
+    <Form>
+      <Menu
+        isOpen={isOpen}
+        onOpen={() => setIsOpen(true)}
+        onClose={() => setIsOpen(false)}
+      >
+        <Menu.Trigger>
+          <Button endIcon={<ChevronDownIcon />} isFullWidth>
+            <Flex alignItems="center" gap="spacingXs">
+              <span
+                css={css`
+                  width: 16px;
+                  height: 16px;
+                  box-shadow: inset 0 0 0 1px rgba(0, 0, 0, 0.1);
+                  background-color: ${value?.value || '#ffffff'};
+                  display: flex;
+                  border-radius: 4px;
+                `}
```

Comment:
In general, I think moving all the CSS styles into a dedicated style file or at least outside of the HTML would make it a bit easier to read 😬